src/main/java/jenkins/advancedqueue/JobGroup.java

```diff
@@ -100,9 +123,40 @@ public void setPriorityStrategy(PriorityStrategy priorityStrategy) {
 	@Deprecated
 	private String jobPattern = ".*";
 	private boolean usePriorityStrategies;
-	private List<JobGroup.PriorityStrategyHolder> priorityStrategies = new ArrayList<JobGroup.PriorityStrategyHolder>();
+	// private List<JobGroup.PriorityStrategyHolder> priorityStrategies = new ArrayList<JobGroup.PriorityStrategyHolder>();
+	private List<PriorityStrategy> priorityStrategies = new ArrayList<PriorityStrategy>();
 
-	private JobGroup() {
+	@DataBoundConstructor
+	public JobGroup(int id, String description, int priority, JobInclusionStrategy jobGroupStrategy, boolean runExclusive, boolean usePriorityStrategies, List<? extends PriorityStrategy> priorityStrategies) {
+		this.id = id;
+		this.description = description;
+		this.priority = priority;
+		this.jobGroupStrategy = jobGroupStrategy;
+		this.runExclusive = runExclusive;
+		this.usePriorityStrategies = usePriorityStrategies;
+		/*
```

Comment:
Copy-pasted from `newInstance()`? Please either cleanup the comment or add a TODO describing why t is commented out. Also, the entire `newIInstance()` might not longer be needed when you have a constructor

Reply:
The `newInstance` method is used by the `PriorityConfiguration.doPriorityConfigSubmit` to serialize the changes done on the /advanced-build-queue configuration page.

At the moment i am missing knowledge on the how this jelly DataBinding works. If this setup is fixed then i guess this method is obsolete.

Reply:
For now i removed the not used full args constructor of the JobGroup class.